src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1422,10 +1426,137 @@ public override IOperation VisitMethodBodyOperation(IMethodBodyOperation operati
         {
             StartVisitingStatement(operation);
 
+            var member = operation.SemanticModel.GetDeclaredSymbol(operation.Syntax);
+            ImmutableArray<IParameterSymbol> parameters = member switch
+            {
+                null => ImmutableArray<IParameterSymbol>.Empty,
+                IFieldSymbol _ => ImmutableArray<IParameterSymbol>.Empty,
+                IParameterSymbol parameterMember => ImmutableArray.Create(parameterMember),
+                IMethodSymbol method => method.Parameters,
+                _ => throw ExceptionUtilities.UnexpectedValue(member)
+            };
+
+            VisitNullChecks(operation, parameters, captureIdForResult);
```

Comment:
>, captureIdForResult [](start = 49, length = 20)

We shouldn't pass this along. In fact, please assert that it's null before calling VisitNullChecks.